vendor/bat-native-ads/src/bat/ads/internal/ads/new_tab_page_ads/new_tab_page_ad.cc

```diff
@@ -44,18 +44,6 @@ void NewTabPageAd::FireEvent(const std::string& uuid,
     return;
   }
 
-  // TODO(https://github.com/brave/brave-browser/issues/14015): Refactor this
-  // logic to only apply frequency capping if the new tab page ad was not served
-  // by the library. |AdServing::MaybeServeAd| is responsible for applying
-  // frequency caps for new tab page ads served by the library
-  new_tab_page_ads::frequency_capping::PermissionRules permission_rules;
-  if (event_type == mojom::NewTabPageAdEventType::kViewed &&
-      !permission_rules.HasPermission()) {
-    BLOG(1, "New tab page ad: Not allowed due to permission rules");
-    NotifyNewTabPageAdEventFailed(uuid, creative_instance_id, event_type);
-    return;
-  }
```

Comment:
This logic is still required if ads are disabled

Reply:
rolled back code deletion

Reply:
Did you also add the condition to check if ads are disabled as this condition should only be checked if ads are disabled

Reply:
yep, condition was added:
https://github.com/brave/brave-core/pull/11945/files#diff-a0bccac6c1f49e72263568f7d661a470ea0673cb351f4060fb817b57a446407fR52